testing/mr_new.cu

```diff
@@ -33,4 +33,16 @@ void TestNewDeleteResourceAlignedAllocation()
         }
     }
 }
+
+void TestNewDeleteResourceEmptyAllocation()
+{
+    thrust::mr::new_delete_resource memres;
+    ASSERT_EQUAL(nullptr, memres.do_allocate(0));
+    memres.do_deallocate(nullptr,0);
+}
+
+
+
 DECLARE_UNITTEST(TestNewDeleteResourceAlignedAllocation);
+DECLARE_UNITTEST(TestNewDeleteResourceEmptyAllocation);
```

Comment:
Make sure that the `DECLARE_UNITTEST` is always the next line after the closing brace of the function it uses to declare the test.